moveit_core/collision_detection_bullet/test/test_bt_collision_detection_panda.cpp

```diff
@@ -0,0 +1,354 @@
+{
+  // Sets the joint values to zero which is a colliding configuration
+  robot_state_->setToDefaultValues();
+  robot_state_->update();
+
+  collision_detection::CollisionRequest req;
+  collision_detection::CollisionResult res;
+  crobot_->checkSelfCollision(req, res, *robot_state_, *acm_);
+  ASSERT_TRUE(res.collision);
+}
+
+/** \brief Two boxes in collision in the world environment. */
+TEST_F(BulletCollisionDetectionTester, WorldToWorldCollision)
+{
+  collision_detection::CollisionRequest req;
+  collision_detection::CollisionResult res;
+
+  shapes::Shape* shape = new shapes::Box(.5, .5, .5);
+  shapes::ShapeConstPtr shape_ptr(shape);
+
+  Eigen::Isometry3d pos_1 = Eigen::Isometry3d::Identity();
+  pos_1.translation().x() = 1;
+  cworld_->getWorld()->addToObject("box", shape_ptr, pos_1);
+
+  Eigen::Isometry3d pos_2 = Eigen::Isometry3d::Identity();
+  pos_2.translation().x() = 1.2;
+  cworld_->getWorld()->addToObject("box_2", shape_ptr, pos_2);
+
+  cworld_->checkRobotCollision(req, res, *crobot_, *robot_state_, *acm_);
+
+  ASSERT_TRUE(res.collision);
+}
+
+/** \brief Adding obstacles to the world which are tested against the robot. Simple cases. */
+TEST_F(BulletCollisionDetectionTester, RobotWorldCollision_1)
+{
+  collision_detection::CollisionRequest req;
+  collision_detection::CollisionResult res;
+
+  shapes::Shape* shape = new shapes::Box(.1, .1, .1);
+  shapes::ShapeConstPtr shape_ptr(shape);
+
+  Eigen::Isometry3d pos1 = Eigen::Isometry3d::Identity();
+  pos1.translation().z() = 0.3;
+  cworld_->getWorld()->addToObject("box", shape_ptr, pos1);
+
+  crobot_->checkSelfCollision(req, res, *robot_state_, *acm_);
+  ASSERT_FALSE(res.collision);
+  res.clear();
+
+  cworld_->checkRobotCollision(req, res, *crobot_, *robot_state_, *acm_);
+  ASSERT_TRUE(res.collision);
+  res.clear();
+
+  cworld_->getWorld()->moveObject("box", pos1);
+  cworld_->checkRobotCollision(req, res, *crobot_, *robot_state_, *acm_);
+  ASSERT_TRUE(res.collision);
+  res.clear();
+
+  cworld_->getWorld()->moveObject("box", pos1);
+  ASSERT_FALSE(res.collision);
+}
+
+/** \brief Adding obstacles to the world which are tested against the robot. */
+TEST_F(BulletCollisionDetectionTester, RobotWorldCollision_2)
+{
+  collision_detection::CollisionRequest req;
+  collision_detection::CollisionResult res;
+  req.max_contacts = 10;
+  req.contacts = true;
+  req.verbose = true;
+
+  shapes::Shape* shape = new shapes::Box(.4, .4, .4);
+  shapes::ShapeConstPtr shape_ptr(shape);
+
+  Eigen::Isometry3d pos1 = Eigen::Isometry3d::Identity();
+  pos1.translation().z() = 0.3;
+  cworld_->getWorld()->addToObject("box", shape_ptr, pos1);
+  cworld_->checkRobotCollision(req, res, *crobot_, *robot_state_, *acm_);
+  ASSERT_TRUE(res.collision);
+  ASSERT_GE(res.contact_count, 3u);
+  res.clear();
+}
+
+/** \brief Continuous self collision checks are not supported yet by tesseract */
+TEST_F(BulletCollisionDetectionTester, DISABLED_ContinuousCollisionSelf)
```

Comment:
Why is this disbled?

Reply:
because continuous to continuous checks are not implemented yet, this feature was / is missing in tesseract and I haven't added it yet. However, I already wrote a test for it.

Reply:
Please add TODO comment with this explanation :-)

Reply:
Added it, not in this PR but in #1551 as CCD got removed from this PR (this test included)